tests/testthat/helper-TealAppDriver.R

```diff
@@ -23,3 +23,34 @@ init_teal_app_driver <- function(...) {
     .package = "teal"
   )
 }
+
+test_plot_changes_no_errors <- function(app_driver, input, value, pws) {
+  plot_before <- app_driver$get_active_module_pws_output(pws)
+  app_driver$set_active_module_input(input, value)
+  app_driver$expect_no_validation_error()
+  testthat::expect_false(identical(plot_before, app_driver$get_active_module_pws_output(pws)))
+}
+
+test_table_changes_no_errors <- function(app_driver, input, value, tws) {
+  table_before <- app_driver$get_active_module_tws_output(tws)
+  app_driver$set_active_module_input(input, value)
+  app_driver$expect_no_validation_error()
+  testthat::expect_false(identical(table_before, app_driver$get_active_module_tws_output(tws)))
+}
+
+test_validation_error <- function(app_driver, input, value = character(0), table = FALSE, message = NULL) {
+  app_driver$set_active_module_input(input, value)
+
+  if (table) {
+    testthat::expect_identical(app_driver$get_active_module_tws_output("table"), data.frame())
+  }
+
+  app_driver$expect_validation_error()
+
+  if (!is.null(message)) {
+    testthat::expect_equal(
+      app_driver$active_module_element_text(sprintf("%s > div > span", input)),
```

Comment:
Can you check if this works? It would be better to use the specific class, instead of the sequence of elements

```suggestion
      app_driver$active_module_element_text(sprintf("%s .shiny-validation-message", input)),
```